### PR TITLE
Render System.Title before trying to trim

### PR DESCRIPTION
**Pull request by kananb**
## Summary of the Pull Request

The [last fix for trimming System.Title](https://github.com/microsoft/onefuzz/pull/3284) wasn't much of a fix as it checked the pre-rendered System.Title length.
This PR fixes that issue by rendering the title in ConstructRenderer() and giving the rendered title a public accessor.

## PR Checklist
* [ ] Applies to work item: #xxx
* [ ] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/onefuzz) and sign the CLI.
* [ ] Tests added/passed
* [ ] Requires documentation to be updated
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Info on Pull Request

System.Title trim fix and some logging that happens when the title is too long.

## Validation Steps Performed

Deployed changes to a manual instance and filed a bug that triggers the updated code.

### CODE REVIEW

File: src/ApiService/ApiService/onefuzzlib/notifications/GithubIssues.cs [hunk 1]
```diff
@@ -3,15 +3,15 @@
...
+    Async.Task GithubIssue(GithubIssuesTemplate config, Container container, IReport reportable, Guid notificationId, Uri instanceUrl);
 }
 
 public class GithubIssues : NotificationsBase, IGithubIssues {
 
     public GithubIssues(ILogger<GithubIssues> logTracer, IOnefuzzContext context)
     : base(logTracer, context) { }
 
-    public async Async.Task GithubIssue(GithubIssuesTemplate config, Container container, IReport reportable, Guid notificationId) {
+    public async Async.Task GithubIssue(GithubIssuesTemplate config, Container container, IReport reportable, Guid notificationId, Uri instanceUrl) {
```
Comment (tevoinea) [inline src/ApiService/ApiService/onefuzzlib/notifications/GithubIssues.cs:14]: You have access to `_context` in this function so we don't need to change the signature. You can get the isntanceUrl the same way you got it in notificationoperations by doing `_context.Creds.GetInstanceUrl()`

File: src/ApiService/ApiService/onefuzzlib/notifications/NotificationsBase.cs [hunk 2]
```diff
@@ -37,6 +37,7 @@ public class Renderer {
         private readonly Report _report;
         private readonly Container _container;
         private readonly string _filename;
+        public string IssueTitle => _issueTitle;
```
Comment (tevoinea) [inline src/ApiService/ApiService/onefuzzlib/notifications/NotificationsBase.cs:40]: Make sure this is public get, private set. We don't want it to be modifiable.

Comment (kananb) [reply src/ApiService/ApiService/onefuzzlib/notifications/NotificationsBase.cs:40]: This declaration only expands to `public string IssueTitle { get; }`, keeping the value immutable from the outside.
image

File: src/ApiService/ApiService/onefuzzlib/notifications/GithubIssues.cs (hunk 1 above)
Comment (kananb) [reply src/ApiService/ApiService/onefuzzlib/notifications/GithubIssues.cs]: Thanks for pointing this out :)

**Review by tevoinea — APPROVED** [review]

**Pull request merged**